algebraic-graphs.cabal

```diff
@@ -94,7 +94,8 @@ library
                         base        >= 4.7     && < 5,
                         containers  >= 0.5.5.1 && < 0.8,
                         deepseq     >= 1.3.0.1 && < 1.5,
-                        mtl         >= 2.1     && < 2.3
+                        mtl         >= 2.1     && < 2.3,
+                        equivalence >= 0.3     && < 0.4
```

Comment:
I'd like to avoid introducing new dependencies.

Could you list the functionality you use along with asymptotic complexity?